src/core/server/plugins/types.ts

```diff
@@ -278,11 +278,94 @@ export interface PluginInitializerContext<ConfigSchema = unknown> {
+   *   constructor(private readonly initContext: PluginInitializerContext) {
+   *     this.logger = initContext.logger.get();
+   *     // `logger` context: `plugins.myPlugin`
+   *     this.mySubLogger = initContext.logger.get('sub'); // or this.logger.get('sub');
+   *     // `mySubLogger` context: `plugins.myPlugin.sub`
+   *   }
+   * }
+   * ```
+   */
   logger: LoggerFactory;
+  /**
+   * Accessors for the plugin's configuration
+   */
   config: {
+    /**
+     * Provide access to Kibana legacy configuration values.
+     *
+     * @remarks Naming not final here, it may be renamed in a near future
+     * @deprecated Accessing configuration values outside of the plugin's config scope is highly discouraged
+     */
     legacy: { globalConfig$: Observable<SharedGlobalConfig> };
+    /**
+     * Return an observable of the plugin's configuration
+     *
+     * @example
+     * ```typescript
+     * // plugins/my-plugin/server/plugin.ts
+     *
+     * export class MyPlugin implements Plugin {
+     *   constructor(private readonly initContext: PluginInitializerContext) {}
+     *   setup(core) {
+     *     this.configSub = this.initContext.config.create<MyPluginConfigType>().subscribe((config) => {
+     *       this.myService.reconfigure(config);
+     *     });
+     *   }
+     *   stop() {
+     *     this.configSub.unsubscribe();
+     *   }
+     * ```
+     *
+     * @example
+     * ```typescript
+     * // plugins/my-plugin/server/plugin.ts
+     *
+     * export class MyPlugin implements Plugin {
+     *   constructor(private readonly initContext: PluginInitializerContext) {}
+     *   async setup(core) {
+     *     this.config = await this.initContext.config.create<MyPluginConfigType>().pipe(take(1)).toPromise();
+     *   }
+     *   stop() {
+     *     this.configSub.unsubscribe();
+     *   }
+     * ```
+     *
+     * @remarks The underlying observable has a replay effect, meaning that awaiting for the first emission
+     *          will be resolved at next tick, without risks to delay any asynchronous code's workflow.
+     */
     create: <T = ConfigSchema>() => Observable<T>;
-    createIfExists: <T = ConfigSchema>() => Observable<T | undefined>;
+    /**
+     * Return the current value of the plugin's configuration synchronously.
+     *
+     * @example
+     * ```typescript
+     * // plugins/my-plugin/server/plugin.ts
+     *
+     * export class MyPlugin implements Plugin {
+     *   constructor(private readonly initContext: PluginInitializerContext) {}
+     *   setup(core) {
+     *     const config = this.initContext.config.get<MyPluginConfigType>();
+     *     // do something with the config
+     *   }
+     * }
+     * ```
+     *
+     * @remarks This should only be used when synchronous access is an absolute necessity, such
+     *          as during the plugin's setup or start lifecycle. For all other usages,
+     *          {@link create} should be used instead.
```

Comment:
Such an irony, considering that we are removing the only config value `elasticsearch.logQueries` supporting reactivity.

Reply:
[YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) 😉